### DISCUSSION TITLE
proably stupid question

### DISCUSSION THREAD

**Discussion by benek984** [auroraresearchlab/netbox-dns#292] [Q&A]
Does netbox-dns plugin actually integrates with your company DNS? Can it read existing and create new enrtries on our Windows Domain controllers?

**Comment by peteeckel** [comment]
They say "there are no dumb questions, just dumb answers" :-)

The philosophy of NetBox (and NetBox DNS as a NetBox plugin) is not to be a user interface for other tools but a "Source of truth". So the straight and simple answer  is "no, it can't, as it wasn't made for that".

However, there are many ways of getting the truth into and out of NetBox (and NetBox DNS, as it inherits these from NetBox):

- A REST API
- A GraphQL API
- Import/Export functionality from and into various formats
- Custom Scripts

With all of these you can import data to and export DNS data from NetBox DNS. At one customer I'm using NetBox as an inventory source for Ansible that also maintains DNS zone files and named configuration files for the DNS infrastructure, another one uses custom scripts to directly dump the zone data to zone files, check them in to git and then provision them to their DNS servers. Import can similarly be done from zone transfers to NetBox DNS using custom scripts. There are many ways of integrating, so many that it is impossible to cater for all needs within NetBox DNS itself (which in consequence is tool agnostic and concentrates on providing RFC conformant, validated and consistent data).

You can look into the `examples` directory for some code that can be used as a base of your own experiments. Nothing too fancy, but you'll see what is possible.

Does all of this make sense to control AD? Probably not. Microsoft have their own user interface for AD DNS which is highly integrated into their solution, and I don't see a point in adding a second one that isn't that well integrated. It might make sense to use NetBox DNS as a "glue tool" between AD and infrastructure not managed within AD, but I don't have a scenario for this in mind - my customers just don't use AD, and neither do I.

Historically I stumbled across NetBox DNS and ultimately started helping developing it because I couldn't find a sensible graphical management tool for DNS data in the open source world and I already was using NetBox, so if you want you can see NetBox DNS (among other things) as an AD GUI replacement for other DNS backends, which might explain some of its properties.

I hope this wasn't a dumb answer, though maybe not the one you were looking for.
